tripal_chado/src/Plugin/Field/FieldFormatter/ChadoOrganismFormatterDefault.php

```diff
@@ -2,37 +2,101 @@
 use Drupal\tripal_chado\TripalField\ChadoFormatterBase;
 
 /**
  * Plugin implementation of default Tripal organism formatter.
  *
  * @FieldFormatter(
  *   id = "chado_organism_formatter_default",
- *   label = @Translation("Chado Organism Reference Formatter"),
- *   description = @Translation("A chado organism reference formatter"),
+ *   label = @Translation("Chado organism formatter"),
+ *   description = @Translation("A chado organism formatter"),
  *   field_types = {
  *     "chado_organism_type_default"
- *   }
+ *   },
+ *   valid_tokens = {
+ *     "[genus]",
+ *     "[genus_abbrev]",
+ *     "[species]",
+ *     "[infratype]",
+ *     "[infratype_abbrev]",
+ *     "[infraname]",
+ *     "[scientific_name]",
+ *     "[abbreviation]",
+ *     "[common_name]",
+ *     "[comment]",
+ *   },
  * )
  */
 class ChadoOrganismFormatterDefault extends ChadoFormatterBase {
 
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    $settings = parent::defaultSettings();
+    $settings['token_string'] = '<i>[genus] [species]</i> [infratype_abbrev] <i>[infraname]</i>';
+    return $settings;
+  }
+
   /**
    * {@inheritdoc}
    */
   public function viewElements(FieldItemListInterface $items, $langcode) {
     $elements = [];
+    $list = [];
+    $token_string = $this->getSetting('token_string');
+
+    // Implement the shortcut token "[scientific_name]". Rather than the overhead
+    // of the chado_get_organism_scientific_name() api call, replicate its behavior
+    // with an equivalent token string.
+    $token_string = preg_replace('/\[scientific_name\]/', $token_string,
+        '[genus] [species] [infratype_abbrev] [infraname]');
 
-    foreach($items as $delta => $item) {
-      $elements[$delta] = [
-        "#markup" => $item->get('label')->getString()
+    foreach ($items as $delta => $item) {
+      $values = [
+        'genus' => $item->get('organism_genus')->getString(),
+        'species' => $item->get('organism_species')->getString(),
+        'infratype' => $item->get('organism_infraspecific_type')->getString(),
+        'infraname' => $item->get('organism_infraspecific_name')->getString(),
+        'abbreviation' => $item->get('organism_abbreviation')->getString(),
+        'common_name' => $item->get('organism_common_name')->getString(),
+        'comment' => $item->get('organism_comment')->getString(),
+      ];
+
+      // Special case handling of abbreviations for genus and infraspecific type
+      $values['genus_abbrev'] = substr($values['genus'], 0, 1) . '.';
+      $values['infratype_abbrev'] = chado_abbreviate_infraspecific_rank($values['infratype']);
+
+      // Substitute values in token string to generate displayed string.
+      $displayed_string = $token_string;
+      foreach ($values as $key => $value) {
+        $displayed_string = preg_replace("/\[$key\]/", $value, $displayed_string);
+      }
+      $list[$delta] = [
+        '#markup' => $displayed_string,
+      ];
+    }
```

Comment:
I'm confused why this code is in the formatter?  We have code built into the field for specifying a token string used to build the label.   If the token string isn't smart enough to build the values shown for the species then perhaps we need to use the `function` action for the field and put this code there.  That way it doesn't matter if you are getting results on the page or if you are getting results via the web services (coming soon).  With this code the user on the web browser will see something different than what is shown in web services.  

I'm not suggestion we fix it here, I just wanted to ask about it and if it sounds like we should use the `function` action on the field to generate this then we can change it later... we just need to add an issue so we don't forget.

Reply:
Also, the `function` action for fields isn't yet written.....  So another reason to leave things as they are here but create an issue to fix this later if need be.

Reply:
This started with the boolean field, since a site administrator may wish to configure how true and false are displayed, e.g. Yes vs. No, Present vs. Absent, etc.
Then it seemed a really simple but helpful way for site admins to configure how linked content is displayed. You may want a displayed value to be different than the title of the entity page itself. This is explained in the now merged PR #1697.
A use case for this could be for a linked analysis, we might want to include the type of analysis, so the token string could be `[name] ([type])`
The default title for analysis is to just display the name.
We can discuss further...

Reply:
@laceysanderson and I had a conversation abou this. I now understand why it is done the way it is.  We had one suggestion though which I'll put on the next comment.